docs/materialize/README.md

```diff
@@ -0,0 +1,288 @@
+specific type of information. Generally for a connector you will have one Endpoint Specification 
+and one or more Resource Specifications. The connector uses this to tell Flow, per binding, what information
+is required to configure that binding.
+
+### Resource Configuration
+This is the information provided by Flow, per binding, to tell it how to configure the connector to
+store that binding.
+
+### Binding
+A binding is a data storage and format configuration. It generally describes where the data comes 
+from (the Collection) and the format in which it is expected to be stored. The field names and types
+are provided as well as any constraints on the fields. 
+
+### Delta Updates
+The Delta Updates flag tells the Flow Runtime if the connector only supports delta changes. In this mode
+the Flow Runtime will not attempt to load any data from the connector and instead just provides the most
+recent version of the document. Essentially the connector works as an append only journal when it runs
+in Delta Updates mode.
+
+### Sharding - KeyBegin/KeyEnd
+In order to facilitate scaling of the Flow ecosystem, document processing is broken into Shards. Chunks of
+documents might be broken into smaller groups for speed of processing, volume of data or both. This is done
+with some sort of sharding function. Essentially if you can take your documents and break them into 
+groups such that all related documents end up in the same group, Flow is able to horizontally scale the
+processing of work. This is done with the Key fields. You might choose to use a date field, IP address, region
+or combinations of all of these for sharding the documents. A document is assigned a Key with a Sharding 
+function and then Flow will use this key value to route documents to ensure documents with the same
+key are all handled by the same pipeline. 
+
+Processing is done using Key ranges where a particular range of keys are associated with an end to end
+pipeline for processing. KeyBegin and KeyEnd determine the range of Keys Handled by a particular pipeline.
+
+### Checkpoints
+To ensure that all data is fully processed, the Flow pipeline uses checkpoints. 
+
+There are two checkpoint types, Driver Checkpoints and Flow Checkpoints. 
+
+Flow Checkpoints are used to delineate pieces of work. Operations are completed using these delineations
+such that if there is a failure, it will start processing again at the last completed checkpoint of data. 
+Flow indicates to a connector that it is starting a particular checkpoint and once the connector confirms the
+checkpoint is complete/committed, Flow will register and consider that a completed piece of work.
+
+Driver Checkpoints are optional values that the connector/driver can register with Flow for tracking state
+information. Flow can provide this information back to the connector if there is a failure once it is restarted.
+Read below for more information about the Driver Checkpoint values.
+
+## RPCs
```

Comment:
I want to clarify some word order/comma related stuff. So, the 3 RPCs are Spec, Validate, and Apply. And this paragraph refers to checks, verifications, setup, streaming. Looking at ordering I want to assume that spec=checks, verification+setup=validate, and streaming=apply? But that doesn't seem right

Reply:
(This is the only thing, from a factual/wording standpoint, that I was confused about in this draft. Will follow up with plans for how I'd like to move forward tomorrow!)

Reply:
The Spec is basically like running the "help" command. 
Validate really does checks/verification
Apply does "setup"
And streaming is when it is actually sending the data which is the last step that goes on while there is still data. 
Hopefully that helps.